src/hpc/serialize.c

```diff
@@ -589,8 +589,8 @@ static void SerializeRecord(Obj obj)
     len = LEN_PREC(obj);
     WriteImmediateObj(INTOBJ_INT(len));
     for (i = 1; i <= len; i++) {
-        UInt rnam = GET_RNAM_PREC(obj, i);
-        Obj  rnams = NAME_RNAM(rnam);
+        Int rnam = GET_RNAM_PREC(obj, i);
+        Obj rnams = NAME_RNAM(rnam >= 0 ? rnam : -rnam);
```

Comment:
Might be getting comment-heavy, but maybe add a comment here saying we save all rnams as positive, so the record will be sorted when loaded back in (I had to go check a few files to convince myself this was the right thing to do, a hint might be nice for future readers)

Reply:
Will do.

Reply:
Actually, we don't store the `rnam` at all, as they can change across sessions, we store the *name* of the RNAM, and for that we must take its absolute value. I'll add comments to clarify this.

Reply:
Ah yes, I misread the method entirely, sorry! However, it's still worth a comment :)

Reply:
I totally agree. Please let me know if you like what I wrote, or some hints what you'd consider better.